pysat/_instrument.py

```diff
@@ -1193,6 +1200,22 @@ def _assign_attrs(self, by_name=False, inst_module=None):
             # No module or name info, default pass functions assigned
             return
 
+        # Check if tag and inst_id are appropriate for the module
+        if inst_id not in inst.inst_ids.keys():
+            inst_id_str = ' '.join([''.join(('"', key, '"'))
+                                    for key in inst.inst_ids.keys()])
```

Comment:
```suggestion
            inst_id_str = ' '.join([''.join(('"', ikey, '"'))
                                    for ikey in inst.inst_ids.keys()])
```

Reply:
Another alternative is to not define `inst_id_str` and `tag_str` here, but instead use: `list(inst.inst_ids.tags.keys()).__repr__()`

Reply:
I merged your suggestions together.